cfme/tests/intelligence/reports/test_canned_corresponds.py

```diff
@@ -93,3 +97,86 @@ def test_cluster_relationships(soft_assert, setup_a_provider):
+        storages.name.label('storages_name'),
+        hosts.name.label('hosts_name')).join(
+            hosts, vms.host_id == hosts.id).join(
+                storages, vms.storage_id == storages.id).filter(
+                    vms.power_state == 'on').order_by(vms.name).all()
+
+    if len(vms_in_db) == len(list(report.data.rows)):
+        for vm in vms_in_db:
+            store_path = '{}/{}'.format(vm.storages_name.encode('utf8'),
+                                        vm.vm_location.encode('utf8'))
+            for item in report.data.rows:
+                if vm.vm_name.encode('utf8') == item['VM Name']:
+                    if (vm.hosts_name.encode('utf8') == item['Host'] and
+                        vm.storages_name.encode('utf8') == item['Datastore'] and
+                        store_path == item['Datastore Path'] and
+                        (str(vm.vm_last_scan).encode('utf8') == item['Last Analysis Time'] or
+                            (str(vm.vm_last_scan).encode('utf8') == 'None' and
+                             item['Last Analysis Time'] == '')
+                         )):
+                            continue
+                    else:
+                        pytest.fail("Found not matching items. db:{} report:{}".format(vm, item))
+    else:
+        pytest.fail("Lenghts of report and BD do not match. db count:{} report count:{}".format(
+            len(vms_in_db), len(list(report.data.rows))))
+
+
+@pytest.mark.tier(3)
+@test_requirements.report
+def test_datastores_summary(soft_assert, setup_a_provider):
+
+    appliance = get_or_create_current_appliance()
+    adb = appliance.db
+    storages = adb['storages']
+    vms = adb['vms']
+    host_storages = adb['host_storages']
+
+    path = ["Configuration Management", "Storage", "Datastores Summary"]
+    report = CannedSavedReport.new(path)
+
+    storages_in_db = adb.session.query(storages.store_type, storages.free_space,
+                                       storages.total_space, storages.name, storages.id).all()
+
+    if len(storages_in_db) == len(list(report.data.rows)):
+        for store in storages_in_db:
+
+            number_of_vms = adb.session.query(vms.id).filter(vms.storage_id == store.id).count()
+            number_of_hosts = adb.session.query(host_storages.host_id).filter(
+                host_storages.storage_id == store.id).count()
+
+            for item in report.data.rows:
+                if store.name.encode('utf8') == item['Datastore Name']:
+                    if (store.store_type.encode('utf8') == item['Type'] and
+                     extract_gb(store.free_space) == float(item['Free Space'].split(' ')[0]) and
+                      extract_gb(store.total_space) == float(item['Total Space'].split(' ')[0]) and
+                       int(number_of_hosts) == int(item['Number of Hosts']) and
+                       int(number_of_vms) == int(item['Number of VMs'])):
+                        continue
```

Comment:
Same as test above, assert/soft_assert the separate fields instead of giant if block + continue.